examples/fast_detect.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright 2024 the Fearless_SIMD Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// Example of fast feature detection.
+
+#![feature(target_feature_11)]
+
+use fearless_simd_macro::simd_dispatch;
+
+// Run this to see the generated asm (using cargo-show-asm):
+//
+// `cargo asm -p fearless_simd --example fast_detect foo 0`
+#[simd_dispatch(levels = "avx2, fallback", detect = "fast")]
+#[inline(never)]
+fn foo(inp: [f32; 4]) -> [f32; 4] {
+    use simd::f32_4::*;
+    let x = inp.into();
+    mul_add(x, x, splat(1.0)).into()
+}
+
+fn main() {
+    // Safety: doing this before launching any threads that might use SIMD.
+    unsafe {
+        fearless_simd::init_simd_detect();
+    }
```

Comment:
worst case you wrap the feature detection in another entry point proc macro, à la 

```rust
#[fearless_simd::main]
fn main() {
...
}
```

Reply:
That opens a potential safety hole, nothing stops you from spawning a thread and applying that macro to a function run by the thread.